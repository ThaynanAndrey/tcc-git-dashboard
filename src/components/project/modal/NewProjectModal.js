import React from 'react';
import BlockUi from 'react-block-ui';

import Modal from '../../layout/Modal';
import ContenteNewProjectModal from './ContentNewProjectModal';
import ProjectModalFooter from './ProjectModalFooter';

/**
 * Stateless component with the function to presenting the New Project Modal.
 * 
 * @author Thaynan Nunes
 */
const NewProjectModal = ({ handleChange, closeModal, createProject, loadingCreateProject }) => {
    return (
        <Modal>
            <BlockUi tag="div" blocking={loadingCreateProject}>
                <ContenteNewProjectModal handleChange={handleChange}/>
                <ProjectModalFooter closeModal={closeModal} createProject={createProject}/>
            </BlockUi>
        </Modal>
    );
}

export default NewProjectModal;