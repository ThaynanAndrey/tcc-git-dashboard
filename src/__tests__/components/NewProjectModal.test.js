import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import NewProjectModal from '../../components/project/modal/NewProjectModal';

describe('Test NewProjectModal', () => {
  it('renders correctly a component to show NewProjectModal', () => {
    const tree = mount(
        <NewProjectModal />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show NewProjectModal with correct props', () => {
    const [handleChange, closeModal, createProject, loadingCreateProject] = new Array(4).fill(jest.fn());

    const tree = mount(
        <NewProjectModal
            handleChange={handleChange} closeModal={closeModal}
            createProject={createProject} loadingCreateProject={loadingCreateProject}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.prop('handleChange')).toEqual(handleChange);
    expect(tree.prop('closeModal')).toEqual(closeModal);
    expect(tree.prop('createProject')).toEqual(createProject);
    expect(tree.prop('loadingCreateProject')).toEqual(loadingCreateProject);
  });
})