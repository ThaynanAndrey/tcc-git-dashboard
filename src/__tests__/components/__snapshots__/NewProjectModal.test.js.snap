// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NewProjectModal renders correctly a component to show NewProjectModal 1`] = `
<NewProjectModal>
  <Modal>
    <div
      className="modal"
    >
      <BlockUi
        loader={[Function]}
        renderChildren={true}
        tag="div"
      >
        <div>
          <ContenteNewProjectModal>
            <div
              className="modal-content"
            >
              <h4>
                Novo Projeto
              </h4>
              <div
                className="row"
              >
                <form
                  className="col s12"
                >
                  <div
                    className="input-field col s5"
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  >
                    <i
                      className="material-icons prefix"
                    >
                      insert_drive_file
                    </i>
                    <input
                      name="projectName"
                      placeholder="Nome Projeto"
                      type="tel"
                    />
                  </div>
                </form>
              </div>
            </div>
          </ContenteNewProjectModal>
          <ProjectModalFooter>
            <div
              className="modal-footer"
            >
              <div
                className="modal-close waves-effect waves-green btn-flat"
              >
                Cancelar
              </div>
              <div
                className="modal-close waves-effect waves-green btn-flat"
              >
                Adicionar
              </div>
            </div>
          </ProjectModalFooter>
          <span />
        </div>
      </BlockUi>
    </div>
  </Modal>
</NewProjectModal>
`;

exports[`Test NewProjectModal renders correctly a component to show NewProjectModal with correct props 1`] = `
<NewProjectModal
  closeModal={[MockFunction]}
  createProject={[MockFunction]}
  handleChange={[MockFunction]}
  loadingCreateProject={[MockFunction]}
>
  <Modal>
    <div
      className="modal"
    >
      <BlockUi
        blocking={[MockFunction]}
        loader={[Function]}
        renderChildren={true}
        tag="div"
      >
        <div
          aria-busy={[MockFunction]}
          className="block-ui undefined"
        >
          <div
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            tabIndex="0"
          />
          <ContenteNewProjectModal
            handleChange={[MockFunction]}
          >
            <div
              className="modal-content"
            >
              <h4>
                Novo Projeto
              </h4>
              <div
                className="row"
              >
                <form
                  className="col s12"
                >
                  <div
                    className="input-field col s5"
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  >
                    <i
                      className="material-icons prefix"
                    >
                      insert_drive_file
                    </i>
                    <input
                      name="projectName"
                      onChange={[MockFunction]}
                      placeholder="Nome Projeto"
                      type="tel"
                    />
                  </div>
                </form>
              </div>
            </div>
          </ContenteNewProjectModal>
          <ProjectModalFooter
            closeModal={[MockFunction]}
            createProject={[MockFunction]}
          >
            <div
              className="modal-footer"
            >
              <div
                className="modal-close waves-effect waves-green btn-flat"
                onClick={[MockFunction]}
              >
                Cancelar
              </div>
              <div
                className="modal-close waves-effect waves-green btn-flat"
                onClick={[MockFunction]}
              >
                Adicionar
              </div>
            </div>
          </ProjectModalFooter>
          <div
            className="block-ui-container"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            tabIndex="0"
          >
            <div
              className="block-ui-overlay"
            />
            <div
              className="block-ui-message-container"
              style={
                Object {
                  "top": undefined,
                }
              }
            >
              <div
                className="block-ui-message"
              >
                <Loader>
                  <div
                    className="loading-indicator"
                  >
                    <span
                      className="loading-bullet"
                    >
                      •
                    </span>
                     
                    <span
                      className="loading-bullet"
                    >
                      •
                    </span>
                     
                    <span
                      className="loading-bullet"
                    >
                      •
                    </span>
                  </div>
                </Loader>
              </div>
            </div>
          </div>
          <span />
        </div>
      </BlockUi>
    </div>
  </Modal>
</NewProjectModal>
`;
